controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -310,8 +319,60 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Update the StorageClassClaim status.
 		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
 
+	} else if r.storageClassClaim.UID != "" {
+
+		// Deletion phase.
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
+
+		// Call `RevokeStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+		// Check if StorageClassClaim is still exists (it might have been manually removed during the StorageClass
+		// removal above).
+		_, err = providerClient.RevokeStorageClassClaim(r.ctx, r.storageCluster.Status.ExternalStorage.ConsumerID, r.storageClassClaim.Name)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Delete StorageClass.
+		// Make sure there are no StorageClass consumers left.
+		// Check if StorageClass is in use, if yes, then fail.
+		// Wait until all PVs using the StorageClass under deletion are removed.
+		// Check for any PVs using the StorageClass.
+		pvList := corev1.PersistentVolumeList{}
+		err := r.list(&pvList, client.MatchingFields{
+			"spec.storageClassName": r.storageClassClaim.Name,
```

Comment:
This is an optional field in the PV spec. Are we sure we are setting it when creating PVs?
Also, if I remember correctly, `storageClassName` does not support all fields, are we sure this selector works?

Reply:
It's being set on creation of PVs.

![image](https://user-images.githubusercontent.com/33557095/166623886-899e4823-608c-4aa7-92f4-de1319b6f82b.png)

> storageClassName does not support all fields

Can you please rephrase, I didn't get this.

Reply:
Ho  sorry I meant `client.MatchingFields` not `storageClassName`. I fixed the original comment